packages/web-api/src/WebClient.ts

```diff
@@ -618,7 +646,8 @@ export class WebClient extends Methods {
     // TODO: better input types - remove any
     const task = () =>
       this.requestQueue.add(async () => {
-        const requestURL = url.startsWith('https' || 'http') ? url : `${this.axios.getUri() + url}`;
+        const requestURL =
+          url.startsWith('https' || 'http') && this.allowAbsoluteUrls ? url : `${this.axios.getUri() + url}`;
```

Comment:
The complexity of this line seems to be growing, could it be more readable it we broke it up with an `if` statement 🤔 ?

Reply:
I was thinking the same... So much is happening here 😓 

I'll explore the `if` statements, but do also like the `const` allowed in the current setup 👾

Reply:
Maybe breaking the logic out into its own function could make sense

```ts
private deriveRequestUrl(url: string): string {
    if (this.allowAbsoluteUrls) {
        return url.startsWith('https' || 'http') ? url : `${this.axios.getUri() + url}`
    }
    return `${this.axios.getUri() + url}`;
}
```

Reply:
Similar logic was moved into a `deriveRequestUrl` function in 5f98b568289def1af5e8049da62e856f87509e3c! I forget `private` functions can be useful sometimes - it's a great call 😅